### PR TITLE
Validator: performing for multiple keys - update rpc proto

### PR DESCRIPTION
**Pull request by shayzluf**
[Part of] #2025 

---

# update rpc proto to support Validator multiple keys

### CODE REVIEW

**Comment by prestonvanloon** [comment]
@shayzluf I fixed your mockgen problem. The script was out of date, sorry about that!

File: scripts/update-mockgen.sh [hunk 1]
```diff
@@ -3,7 +3,10 @@
...
 # Use a space to separate mock destination from its interfaces.
 
-mocks=("./validator/internal/attester_service_mock.go AttesterServiceClient"
+mocks=(
+      "./beacon-chain/internal/beacon_service_mock.go BeaconServiceServer,BeaconService_LatestAttestationServer,BeaconService_WaitForChainStartServer"
+      "./beacon-chain/internal/validator_service_mock.go ValidatorServiceServer,ValidatorService_WaitForActivationServer"
+      "./validator/internal/attester_service_mock.go AttesterServiceClient"
        "./validator/internal/beacon_service_mock.go BeaconServiceClient,BeaconService_LatestAttestationClient,BeaconService_WaitForChainStartClient"
        "./validator/internal/proposer_service_mock.go ProposerServiceClient"
        "./validator/internal/validator_service_mock.go ValidatorServiceClient")
```
Comment (prestonvanloon) [inline scripts/update-mockgen.sh]:
```suggestion
       "./validator/internal/validator_service_mock.go ValidatorServiceClient,ValidatorServiceClient_WaitForActivationClient")
```

File: beacon-chain/rpc/validator_server_test.go [hunk 2]
```diff
@@ -70,8 +70,9 @@ func TestNextEpochCommitteeAssignment_WrongPubkeyLength(t *testing.T) {
 	validatorServer := &ValidatorServer{
 		beaconDB: db,
 	}
-	req := &pb.ValidatorEpochAssignmentsRequest{
-		PublicKey:  []byte{},
+	req := &pb.CommitteeAssignmentsRequest{
+
```
Comment (prestonvanloon) [inline beacon-chain/rpc/validator_server_test.go]: remove this new line

File: beacon-chain/rpc/validator_server_test.go [hunk 3]
```diff
@@ -92,13 +93,13 @@ func TestNextEpochCommitteeAssignment_CantFindValidatorIdx(t *testing.T) {
...
-		PublicKey:  pubKey,
+	req := &pb.CommitteeAssignmentsRequest{
+		PublicKey:  [][]byte{pubKey},
 		EpochStart: params.BeaconConfig().GenesisEpoch,
 	}
-	want := fmt.Sprintf("validator %#x does not exist", req.PublicKey)
+	want := fmt.Sprintf("validator %#x does not exist", req.PublicKey[0])
 	if _, err := vs.CommitteeAssignment(context.Background(), req); !strings.Contains(err.Error(), want) {
-		t.Errorf("Expected %v, received %v", want, err)
+		t.Errorf("Expected %v, received %v", want, err.Error())
```
Comment (prestonvanloon) [inline beacon-chain/rpc/validator_server_test.go]:
```suggestion
		t.Errorf("Expected %v, received %v", want, err)
```

File: proto/beacon/rpc/v1/services.proto [hunk 4]
```diff
@@ -128,22 +128,28 @@ message ValidatorIndexResponse {
...
+message CommitteeAssignmentsRequest {
     uint64 epoch_start = 1;
-    bytes public_key = 2;
+    repeated bytes public_key = 2;
 }
 
 message PendingDepositsResponse {
     repeated ethereum.beacon.p2p.v1.Deposit pending_deposits = 1;
 }
-
```
Comment (prestonvanloon) [inline proto/beacon/rpc/v1/services.proto]: please keep space between messages

File: proto/beacon/rpc/v1/services.proto [hunk 5]
```diff
@@ -128,22 +128,28 @@ message ValidatorIndexResponse {
...
+        repeated uint64 committee = 1;
+        uint64 shard = 2;
+        uint64 slot = 3;
+        bool is_proposer = 4;
+        bytes public_key = 5;
+        ValidatorStatus status = 6;
+    }
 }
 
+
```
Comment (prestonvanloon) [inline proto/beacon/rpc/v1/services.proto]: remove this newline

File: validator/client/validator_test.go [hunk 6]
```diff
@@ -8,13 +8,14 @@ import (
 	"testing"
 	"time"
 
+	"github.com/prysmaticlabs/prysm/validator/internal"
+
```
Comment (prestonvanloon) [inline validator/client/validator_test.go]: remove newline

File: beacon-chain/rpc/validator_server_test.go (hunk 2 above)
Comment (shayzluf) [reply beacon-chain/rpc/validator_server_test.go]: done

**Comment by prestonvanloon** [comment]
Please wait to merge this until i can run in a cluster deployment for sanity testing.

**Pull request merged**